mtp_cashbook/apps/cashbook/tests/test_forms.py

```diff
@@ -0,0 +1,122 @@
+
+    def setUp(self):
+        super(ProcessTransactionBatchFormTestCase, self).setUp()
+        self.user = mock.MagicMock()
+        self.request = mock.MagicMock()
+
+        self.pending_transactions_response = {
+            'count': 4,
+            'results': [{'id': i} for i in range(1, 5)]
+        }
+
+    @mock.patch('cashbook.forms.get_connection')
+    def __call__(self, result, mocked_get_connection, *args, **kwargs):
+        self.mocked_get_connection = mocked_get_connection
+        super(ProcessTransactionBatchFormTestCase, self).__call__(result, *args, **kwargs)
+
+    def test_init_form_with_already_locked_transactions(self):
+        self.mocked_get_connection().\
+            transactions()().\
+            get.return_value = self.pending_transactions_response
+
+        form = ProcessTransactionBatchForm(self.request)
+
+        expected_choices = [
+            (t['id'], t) for t in self.pending_transactions_response['results']
+        ]
+        self.assertEqual(form.transaction_choices, expected_choices)
+        self.assertEqual(form.fields['transactions'].choices, expected_choices)
+
+    def test_init_form_with_no_locked_transactions(self):
+        self.mocked_get_connection().transactions()().get.side_effect = [
+            {
+                'count': 0,
+                'results': 0
+            },
+        ] + [self.pending_transactions_response]
+
+        form = ProcessTransactionBatchForm(self.request)
+        expected_choices = [
+            (t['id'], t) for t in self.pending_transactions_response['results']
+        ]
+        self.assertEqual(form.transaction_choices, expected_choices)
+        self.assertEqual(form.fields['transactions'].choices, expected_choices)
+
+    def test_credit_some(self):
+        """
+        Tests that [2, 3] gets credited and [1, 4] discarded
+        """
+        self.mocked_get_connection().\
+            transactions()().\
+            get.return_value = self.pending_transactions_response
+
+        to_credit = ['2', '3']
+        form = ProcessTransactionBatchForm(
+            self.request,
+            data={
+                'transactions': to_credit
+            }
+        )
+        self.assertTrue(form.is_valid())
+        form.save()
+
+        # checking call to patch
+        mocked_patch = form.client.transactions()().patch
+        self.assertEqual(mocked_patch.call_count, 1)
+        credited_transactions = mocked_patch.call_args[0][0]
+        self.assertListEqual(
+            sorted([t['id'] for t in credited_transactions]),
+            sorted(to_credit)
+        )
+
+        # checking call to release
+        mocked_release = form.client.transactions()().release.post
+        self.assertEqual(mocked_release.call_count, 1)
+        discarded_transactions = mocked_release.call_args[0][0]
+        self.assertListEqual(
+            sorted(discarded_transactions['transaction_ids']),
+            sorted(['1', '4'])
+        )
```

Comment:
It would be really confusing for me to have different tests as we are testing one global thing: given a set of ids, some should get credited and the rest should get discarded.

Also, `test_release_after_partial_credit_only_releases_uncredited_transactions` would have to check that the credit bit worked anyway so it wouldn't be better than this.
Also if the credit bit broke `test_release_after_partial_credit_only_releases_uncredited_transactions` would break and it would be even harder to understand what happened.